SL/Controller/Order.pm

```diff
@@ -1877,27 +1820,53 @@ sub make_order {
   # be retrieved via items until the order is saved. Adding empty items to new
   # order here solves this problem.
   my $order;
-  $order   = SL::DB::Order->new(id => $::form->{id})->load(with => [ 'orderitems', 'orderitems.part' ]) if $::form->{id};
-  $order ||= SL::DB::Order->new(orderitems  => [],
-                                record_type => $::form->{type},
-                                currency_id => $::instance_conf->get_currency_id(),);
+  if ($::form->{id}) {
+    $order   = SL::DB::Order->new(
+      id => $::form->{id}
+    )->load(
+      with => [
+        'orderitems',
+        'orderitems.part',
+      ]
+    );
+  } else {
+    $order = SL::DB::Order->new(
+      orderitems  => [],
+      record_type => $::form->{type},
+      currency_id => $::instance_conf->get_currency_id(),
+    );
+    $order = SL::Model::Record->update_after_new($order)
+  }
 
   my $cv_id_method = $order->type_data->properties('customervendor'). '_id';
   if (!$::form->{id} && $::form->{$cv_id_method}) {
     $order->$cv_id_method($::form->{$cv_id_method});
     $order = SL::Model::Record->update_after_customer_vendor_change($order);
   }
 
-  my $form_orderitems                  = delete $::form->{order}->{orderitems};
-  my $form_periodic_invoices_config    = delete $::form->{order}->{periodic_invoices_config};
+  # don't assign hashes as objects
+  my $form_orderitems               = delete $::form->{order}->{orderitems};
+  my $form_periodic_invoices_config = delete $::form->{order}->{periodic_invoices_config};
 
   $order->assign_attributes(%{$::form->{order}});
 
+  # restore form values
+  $::form->{order}->{orderitems}               = $form_orderitems;
```

Comment:
Dieses Wiederherstellen der form (orderitems) sorgt leider dafür, dass gelöschte Positionen eines schon gespeicherten Auftrags, von dem aus inline ein Artikel erzeugt wird, danach wieder auftauchen.
Nachstellbar: Auftrag mit  Artikel(n) anlegen, speichern. Artikel löschen. Artikel inline erzeugen und zurück. Gelöscthe Artikel sind wieder da.

Analog vermutlich in Lieferschein u Reklamation

Reply:
Das Problem lag im Aufruf der Funktion `$self->action_edit()`. Diese hat den gespeicherten Beleg geladen und vergleicht ihn nicht mit der geänderten Form.